packages/webview_flutter/webview_flutter_android/lib/src/android_webview.dart

```diff
@@ -66,6 +66,91 @@ class WebView {
+  ///
+  /// The [encoding] parameter specifies whether the data is base64 or URL
+  /// encoded. If the data is base64 encoded, the value of the encoding
+  /// parameter must be `'base64'`. HTML can be encoded with
+  /// `base64.encode(bytes)` like so:
+  /// ```dart
+  /// import 'dart:convert';
+  ///
+  /// final unencodedHtml = '''
+  ///   <html><body>'%28' is the code for '('</body></html>
+  /// ''';
+  /// final encodedHtml = base64.encode(utf8.encode(unencodedHtml));
+  /// print(encodedHtml);
+  /// ```
+  ///
+  /// The [mimeType] parameter specifies the format of the data. If WebView
+  /// can't handle the specified MIME type, it will download the data. If
+  /// `null`, defaults to 'text/html'.
+  Future<void> loadData(String data, String? mimeType, String? encoding) {
+    return api.loadDataFromInstance(
+      this,
+      data,
+      mimeType ?? _nullStringIdentifier,
+      encoding ?? _nullStringIdentifier,
+    );
+  }
+
+  /// Loads the given data into this WebView, using baseUrl as the base URL for
+  /// the content.
+  ///
+  /// The base URL is used both to resolve relative URLs and when applying
+  /// JavaScript's same origin policy. The [historyUrl] is used for the history
+  /// entry.
+  ///
+  /// The [mimeType] parameter specifies the format of the data. If WebView
+  /// can't handle the specified MIME type, it will download the data. If
+  /// `null`, defaults to 'text/html'.
+  ///
+  /// Note that content specified in this way can access local device files (via
+  /// 'file' scheme URLs) only if baseUrl specifies a scheme other than 'http',
+  /// 'https', 'ftp', 'ftps', 'about' or 'javascript'.
+  ///
+  /// If the base URL uses the data scheme, this method is equivalent to calling
+  /// [loadData] and the [historyUrl] is ignored, and the data will be treated
+  /// as part of a data: URL, including the requirement that the content be
+  /// URL-encoded or base64 encoded. If the base URL uses any other scheme, then
+  /// the data will be loaded into the WebView as a plain string (i.e. not part
+  /// of a data URL) and any URL-encoded entities in the string will not be
+  /// decoded.
+  ///
+  /// Note that the [baseUrl] is sent in the 'Referer' HTTP header when
+  /// requesting subresources (images, etc.) of the page loaded using this
+  /// method.
+  ///
+  /// If a valid HTTP or HTTPS base URL is not specified in [baseUrl], then
+  /// content loaded using this method will have a `window.origin` value of
+  /// `"null"`. This must not be considered to be a trusted origin by the
+  /// application or by any JavaScript code running inside the WebView (for
+  /// example, event sources in DOM event handlers or web messages), because
+  /// malicious content can also create frames with a null origin. If you need
+  /// to identify the main frame's origin in a trustworthy way, you should use a
+  /// valid HTTP or HTTPS base URL to set the origin.
+  Future<void> loadDataWithBaseUrl(
```

Comment:
This method could also use optional named parameters with `data` being `@required`.

Reply:
Thanks for thinking along, updated as you suggested.